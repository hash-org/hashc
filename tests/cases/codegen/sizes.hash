// stage=exe, run=pass

printing := import("printing");


Array := struct(
    data: [u8],
    capacity: usize,
)

Decimal := struct(
    sign: u8,
    exponent: i32,
    mantissa: [u8; 16],
)

main := () => {
    printing::print_usize(#size_of Decimal); // Should be 24
    println("")
    printing::print_usize(#size_of u32); // Should be 4
    println("")
    printing::print_usize(#size_of str); // Should be 8
    println("")
    printing::print_usize(#size_of Array); // Should be 24
    println("")
}
