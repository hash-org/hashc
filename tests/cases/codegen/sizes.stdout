24
4
16
24
